Jenkinsfile

```diff
@@ -63,110 +69,196 @@ pipeline {
+              }
+              currentBuild.description = (params.RELEASE_DRY_RUN) ? "${env.DOCKER_TAG} & ${env.SECONDARY_DOCKER_TAG} - DRY RUN" : "${env.DOCKER_TAG} & ${env.SECONDARY_DOCKER_TAG}"
+            }
           }
         }
       }
       post {
         success {
           script {
-            if (params.RELEASE_BUILD && !params.RELEASE_DRY_RUN) {
-              Map buildInfoServer = [
-                "project_id": service_server,
-                "build_id": env.DOCKER_TAG,
-                "release_id": params.RELEASE_NAME,
-                "git_tag": env.GIT_TAG
-              ]
-              createBuild(
-                buildInfoServer,
-                rmsSupportedEnvironment.production
-              )
+            if (isHotfix) {
+              env.GIT_TAG = releaseTag.create(env.HOTFIX_NAME)
+            } else if (params.RELEASE_BUILD && !params.RELEASE_DRY_RUN) {
+              env.GIT_TAG = releaseTag.getFormattedTag(params.RELEASE_NAME)
+              if (releaseTag.doesReleaseTagExist(params.RELEASE_NAME)) {
+                echo "REBUILD: Tag ${env.GIT_TAG} already exists"
+                return
+              }
+              if (params.RELEASE_DRY_RUN) {
+                echo "DRY RUN: Would have created ${env.GIT_TAG} and pushed it to the repo"
+                return
+              }
+              releaseTag.create(params.RELEASE_NAME)
             }
           }
         }
       }
     }
-    stage('Dockerize Secondary') {
+    stage('Publish') {
       when {
-        not { expression { isPr } }
+        allOf {
+          not { expression { isPr } }
+          not { expression { skip } }
+          not { expression { return params.RELEASE_BUILD && params.RELEASE_DRY_RUN } }
+        }
       }
-      steps {
-        script {
-          lastStage = env.STAGE_NAME
-          // Here's where we're getting the secondary artifact (named
-          // via env.STORE_PG_ARTIFACT) out for dockerizing, per the
-          // comment above.
-          if (params.RELEASE_BUILD) {
-            env.REGISTRY_PUSH = (params.RELEASE_DRY_RUN) ? 'none' : 'all'
-            env.SECONDARY_DOCKER_TAG = dockerize_secondary.docker_build_specify_tag_and_push(params.RELEASE_NAME, sbtbuild.getDockerPath(project_wd_secondary), env.STORE_PG_ARTIFACT, env.REGISTRY_PUSH)
-          } else {
-            env.REGISTRY_PUSH = 'internal'
-            env.SECONDARY_DOCKER_TAG = dockerize_secondary.docker_build('STAGING', env.GIT_COMMIT, sbtbuild.getDockerPath(project_wd_secondary), env.STORE_PG_ARTIFACT, env.REGISTRY_PUSH)
+      stages {
+        stage('Publish - SoQL Server') {
+          steps {
+            script {
+              lastStage = env.STAGE_NAME
+              env.REGISTRY_PUSH = (isHotfix || params.RELEASE_BUILD) ? 'all' : 'internal'
+              dockerize_server.specify_tag_and_push(env.DOCKER_TAG, dockerize_server.getTag(), env.REGISTRY_PUSH)
```

Comment:
The only methods we currently have for publishing an existing docker image, first tag the image.  I made a ticket to make this better in the future: EN-65752.  For now, I've made that action redundant by passing in the tag that's already been set.